remove auth and other headers from api configuration

paritosh-08 opened the pull request:
This PR removes auth and other headers from the API Configuration struct. This PR also adds some documentation around how we can add headers to be sent in API requests.

paf31 commented on ndc-client/src/apis/configuration.rs:
```diff
@@ -1,11 +1,14 @@
 pub struct Configuration {
     pub base_path: String,
     pub user_agent: Option<String>,
     pub client: reqwest::Client,
```
`client` provides a way to edit the default set of headers. Why not use that?

paritosh-08 replied:
Are you talking about [`ClientBuilder`](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html)?

So, the Configuration will have the client (already configured with default headers). While initializing, the user will have to make sure that the appropriate headers are already part of the client.

I am okay with this change, the only downside is that while initializing, we will have to create a [`ClientBuilder`](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html) first, then [add default headers](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.default_headers) and then create client (and handle errors as the [`ClientBuilder.build`](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.build) returns `Result<Client>`).

paf31 commented on ndc-client/src/apis/configuration.rs:
```diff
@@ -1,11 +1,14 @@
     pub basic_auth: Option<BasicAuth>,
     pub oauth_access_token: Option<String>,
-    pub bearer_access_token: Option<String>,
+    // TODO: should we remove oauth access tokens and basic_auth as well?
```
Yes I think so.

paritosh-08 replied:
Done

paritosh-08 commented:
@paf31 I have removed the headers.

I have also added some documentation around how the client is supposed to be built with some default set of headers. Can you please review that as well?

paf31 commented on ndc-client/src/apis/configuration.rs:
```diff
@@ -1,11 +1,42 @@
+/// Configuration for the API.
```
```suggestion
/// Configuration for the API client
```

paritosh-08 replied:
Thanks, done

paf31 commented on ndc-client/src/apis/configuration.rs:
```diff
@@ -1,11 +1,42 @@
-    pub basic_auth: Option<BasicAuth>,
-    pub oauth_access_token: Option<String>,
-    pub bearer_access_token: Option<String>,
     pub api_key: Option<ApiKey>,
```
Let's remove this one too while we're at it, please.

paritosh-08 replied:
Done

paf31 reviewed (approved):
Looks good, please feel free to merge after removing `api_key` and `ApiKey`.

paf31 reviewed (approved):
Thanks!
